src/views/editor/mod.rs

```diff
@@ -1093,12 +1080,16 @@ impl TextLayoutProvider for Editor {
             line_content_original.to_string()
         };
         // Combine the phantom text with the line content
-        let phantom_text = doc.phantom_text(edid, style.as_ref(), line);
+        let phantom_text = doc.phantom_text(edid, &self.editor_style.get_untracked(), line);
         let line_content = phantom_text.combine_with_text(&line_content);
 
         let family = style.font_family(edid, line);
         let attrs = Attrs::new()
-            .color(style.color(edid, EditorColor::Foreground))
+            .color(
+                self.editor_style
+                    .with(|s| s.text_color())
+                    .unwrap_or(Color::BLACK),
```

Comment:
Just have `text_color` return a `Color` and unwrap to default itself

Reply:
You could also just get-tracked editor style once at the top to make the function cleaner

Reply:
TextColor is a floem property and it's an Option<Color>. I'm not really sure why though

> You could also just get-tracked editor style once at the top to make the function cleaner

`with` avoids cloning the style struct

Reply:
> TextColor is a floem property and it's an Option<Color>. I'm not really sure why though

To allow you to know whether you can use a default presumably? (Though probably there's some way to just go "use this as the base style for my view")

Could still just do an `impl EditorStyle { fn ed_text_color() ..`

> `with` avoids cloning the style struct

and `EditorStyle` doesn't have that much in it. Though I don't consider this style too bad.

Reply:
> and EditorStyle doesn't have that much in it. Though I don't consider this style too bad.

The style structs are surprisingly large. `EditorStyle` is 536 bytes and they grow quickly with more fields. 

> Though I don't consider this style too bad.

I also don't think the `with` adds too much clutter

> Could still just do an impl EditorStyle { fn ed_text_color() ..

👍